src/pages/CartPage.js

```diff
@@ -0,0 +1,50 @@
+import { Fragment } from "react";
+import CardCart from "../componnets/CardCart";
+import classes from "./CartPage.module.css";
+import { useState } from "react";
+
+const CartPage = ({
+  productsOnCart,
+  deleteProduct,
+  editProductAmount,
+  resetProducts,
+}) => {
+  const [isProcessing, setIsProcessing] = useState(false);
+  //calculate total amount to pay
+  const total = productsOnCart.reduce((current, total) => {
+    return current + total.price * total.amount;
+  }, 0);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    resetProducts([]);
```

Comment:
A "reset" function, especially a prop, should not take any arguments. Resetting means resetting and it's not the business of the cart page component how the initial value looks like, that it resets to. The empty array should be set as state only in the `App` component and every other component should just have a prop `onReset` or `onResetCart` that can be called whenever the user resets the cart.